src/builder/command.rs

```diff
@@ -3782,7 +3788,9 @@ impl Command {
 
             self._propagate();
             self._check_help_and_version();
-            self._propagate_global_args();
+            if !self.is_set(AppSettings::DisablePropagatedArgs) {
+                self._propagate_global_args();
+            }
```

Comment:
Why do we need to disable global args?

Reply:
An example is `cmd help subcmd <TAB>` and it might give the completions `subsubcmd` and `--some-global`. Please correct me if I'm wrong, but I thought it wouldn't make sense to keep any args there, global or not, as the help subcommand will ignore those when running the actual CLI command.

Reply:
Ok, this is fixing an existing bug.  Reproduction case:
```rust
#!/usr/bin/env -S rust-script --debug

//! ```cargo
//! [dependencies]
//! clap = { version = "3.2", features = ["derive"] }
//! ```

use clap::Parser;

#[derive(Parser)]
struct Config {
    #[clap(short, long, global = true)]
    verbose: bool,

    #[clap(subcommand)]
    command: Command,
}

#[derive(clap::Subcommand)]
enum Command {
    One,
    Two,
}

fn main() {
    // Be sure to enable the 'derive' feature of clap in Cargo.toml
    let _ = Config::parse();
}
```
yields
```
$ ./clap-4115.rs help help
clap-4115_eb4d3f07a388d728529d994c-help
Print this message or the help of the given subcommand(s)

USAGE:
    clap-4115_eb4d3f07a388d728529d994c help [OPTIONS] [SUBCOMMAND]...

ARGS:
    <SUBCOMMAND>...    The subcommand whose help message to display

OPTIONS:
    -v, --verbose
```


Can we split this into its own PR with tests?  We need to consider the implications of this without being lost in the discussion on the wider completions issue.

Reply:
Also, in doing so, I wonder about the trade offs of instead checking `is_help_subcommand_disabled` and if the command name is `help` rather than adding a new setting.

Reply:
Sure, so in summary, the plan is for me to revert the special handling of global args in this PR, and this PR should then be ready to merge right?

And then a followup PR dealing with the global args in both completions case and in the help message, plus anywhere else that potentially comes up. And agreed, I'd prefer solutions that don't require adding settings flags.

Reply:
Sorry, hadn't finished looking at the new changes (was unsure which of the two you'd prefer to do first)

Reply:
Alright, I think I've completed my feedback pending this being split into two PRs